inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -0,0 +1,115 @@
+= Releasing INCEpTION
+
+== Prerequisites
+
+When releasing and publishing the resulting artifact, make sure that you have actually the rights
+to publish new artifacts to the target repository. See e.g.
+https://maven.apache.org/settings.html#Servers[the Maven documentationon authentication] or
+https://dkpro.github.io/dkpro-core/pages/setup-maven[DKPro Core - Setting up Maven repository]
+for more information on how to do this. Additionally, it assumes that you have write permissions
+to the protected `master` branch and to the protected maintenance branches.
+
+== Branching model
+
+INCEpTION uses a branching model where the development towards the next feature release happens on the
+`master`. Additionally, there are *maintenance* (also called *stable*) branches for all past feature 
+releases which are used to create bugfix releases when necessary. 
+
+To prepare a new feature release, we first create a maintenance branch from `master`. Assuming the current
+development version on the `master` branch is `0.8.0-SNAPSOHT`, we create a maintenance branch by the name
+`0.8.x`. After this, the version on the master branch must be manually updated to the next feature version,
+typically by increasing the second digit of the version (e.g. `0.9.0-SNAPSOHT` becomes `0.9.0-SNAPSOHT`).
+
+The process of doing the initial feature release on a new maintenance branch (e.g. for version `0.8.0`) is 
+largely the same then as for performing a bugfix release (e.g. for version `0.8.1`).
```

Comment:
...performing a bugfix release (e.g. for version `0.8.1`) as described in the next section.